src/Symfony/Component/RateLimiter/LimiterFactory.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+use Symfony\Component\Lock\LockFactory;
+use Symfony\Component\Lock\NoLock;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\RateLimiter\Storage\StorageInterface;
+
+/**
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @final
+ */
+class LimiterFactory
+{
+    private $config;
+    private $storage;
+    private $lockFactory;
+
+    public function __construct(array $config, StorageInterface $storage, ?LockFactory $lockFactory)
+    {
+        $this->storage = $storage;
+        $this->lockFactory = $lockFactory;
+
+        $options = new OptionsResolver();
+        self::configureOptions($options);
+
+        $this->config = $options->resolve($config);
+    }
+
+    public function createLimiter(?string $key = null): LimiterInterface
```

Comment:
might be `create()` instead (`Limiter::create()` reads better than `LimiterFactory::createLimiter()`)